src/lib.rs

```diff
@@ -125,14 +125,60 @@ use core::ptr;
 pub struct Removing<'a, T> {
     // Type invariants:
     // - vec.capacity() >= len
-    // - vec[..slot] is initialized (note: dye to safety guarantees vec.len is
-    //   set to 0, so this is not 'real' indexing)
+    // - vec[..slot] is initialized (note: due to safety guarantees vec.len is set to 0, so this is
+    //   not 'real' indexing)
     // - vec[curr..len] is initialized
     // - vec[..len] is the same allocation
     vec: &'a mut Vec<T>,
     slot: usize,
     curr: usize,
     len: usize,
+    /* Example of how this works, step-by-step:
```

Comment:
I think it makes sense to put this explanatory comment outside of struct definition